bridger-cli/src/utils/transactionHandler.ts

```diff
@@ -0,0 +1,159 @@
+}
+
+/**
+ * Handles transactions for a given veaOutbox and epoch.
+ *
+ * @param chainId - The chainId of veaOutbox chain
+ * @param epoch - The epoch number for which the transactions are being handled
+ * @param veaOutbox - The veaOutbox instance to use for sending transactions
+ * @param claim - The claim object for the epoch
+ * @returns An instance of the TransactionHandler class
+ *
+ * @example
+ * const txHandler = new TransactionHandler(11155111, 240752, veaOutbox, claim);
+ * txHandler.sendTransaction(txData);
+ */
+
+export class TransactionHandler {
+  public epoch: number;
+  public veaOutbox: any;
+  public chainId: number;
+  public claim: ClaimStruct | null;
+  public getBridgeConfig: typeof getBridgeConfig;
+  public requiredConfirmations: number = 12;
+
+  public pendingTransactions: PendingTransactions = {
+    claim: null,
+    verifySnapshot: null,
+    withdrawClaimDeposit: null,
+    startVerification: null,
+  };
+
+  constructor(
+    chainId: number,
+    epoch: number,
+    veaOutbox: any,
+    claim?: ClaimStruct,
+    fetchBridgeConfig: typeof getBridgeConfig = getBridgeConfig
+  ) {
+    this.epoch = epoch;
+    this.veaOutbox = veaOutbox;
+    this.chainId = chainId;
+    this.claim = claim;
+    this.getBridgeConfig = fetchBridgeConfig;
+  }
+
+  public async checkTransactionPendingStatus(trnxHash: string | null): Promise<boolean> {
+    if (trnxHash == null) {
+      return false;
+    }
+
+    const receipt = await this.veaOutbox.provider.getTransactionReceipt(trnxHash);
+
+    if (!receipt) {
+      console.log(`Transaction ${trnxHash} is pending`);
+      return true;
+    }
+
+    const currentBlock = await this.veaOutbox.provider.getBlock("latest");
+    const confirmations = currentBlock.number - receipt.blockNumber;
+
+    if (confirmations >= this.requiredConfirmations) {
+      console.log(`Transaction ${trnxHash} is final with ${confirmations} confirmations`);
+      return false;
+    } else {
+      console.log(`Transaction ${trnxHash} is not final yet.`);
+      return true;
+    }
+  }
+
+  public async makeClaim(stateRoot: string) {
+    if (await this.checkTransactionPendingStatus(this.pendingTransactions.claim)) {
+      console.log("Claim transaction is still pending with hash: " + this.pendingTransactions.claim);
+      return;
+    }
+    const bridgeConfig = this.getBridgeConfig(this.chainId);
+    const estimateGas = await this.veaOutbox.estimateGas.claim(this.epoch, stateRoot, { value: bridgeConfig.deposit });
+    const claimTransaction = await this.veaOutbox.claim(this.epoch, stateRoot, {
+      value: bridgeConfig.deposit,
+      gasLimit: estimateGas,
+    });
+    console.log(`Epoch ${this.epoch} was claimed with trnx hash ${claimTransaction.hash}`);
+    this.pendingTransactions.claim = claimTransaction.hash;
+  }
+
+  public async startVerification(latestBlockTimestamp: number) {
+    if (this.claim == null) {
+      throw new Error("Claim is not set");
```

Comment:
You use this error in more than one place, it would be nice to have a custom one so the responsibility is only present once and make tests assertions easier